core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/DatabricksParseHelper.scala

```diff
@@ -108,4 +115,41 @@ object DatabricksParseHelper extends Logging {
         Map.empty
     }
   }
+
+  /**
+   * Maps the Photon operator names to Spark operator names using a mapping JSON file.
+   */
+  private lazy val photonToSparkMapping: Map[String, String] = {
+    val jsonString = UTF8Source.fromResource(PHOTON_OPERATOR_MAPPING_FILE).mkString
+    val json = JsonMethods.parse(jsonString)
+    // Implicitly define JSON formats for deserialization using DefaultFormats
+    implicit val formats: Formats = DefaultFormats
+    // Extract and deserialize the JValue object into a Map[String, String]
+    // TODO: Instead of only extracting the first value, we should consider extracting all values
+    json.extract[Map[String, List[String]]].mapValues(_.head)
+  }
+
+  /**
+   * Replaces a Photon node name in the SparkPlanInfo object with a corresponding Spark node name,
+   * if a mapping exists.
+   */
+  def processPhotonPlan(planInfo: SparkPlanInfo): Option[SparkPlanInfo] = {
+    // Check if the node name contains a Photon node
+    val photonNodeOpt = PHOTON_PATTERN.findFirstIn(planInfo.nodeName)
+    // Early return if node is not a Photon node
+    if (photonNodeOpt.isEmpty) {
+      return None
+    }
+    // If a Photon node is found, try to map it to a Spark node
+    photonNodeOpt.flatMap(photonToSparkMapping.get).map { sparkNode =>
+      // Create a new SparkPlanInfo object with node name and description replaced
+      // with the Spark node name
+      new SparkPlanInfo(
+        nodeName = sparkNode,
+        simpleString = planInfo.simpleString.replace(planInfo.nodeName, sparkNode),
+        children = planInfo.children,
+        metadata = planInfo.metadata,
```

Comment:
does the meta all lineup and metrics?  I wouldn't expect metrics to be anywhere close

Reply:
Could you explain the lining up and metrics?

Reply:
lets talk about off line.  I mean that if photon execs have certain parameters, we need to make sure the CPU one we replace it with has its parameters filled in correctly.  In addition to just parameters there are metrics that come out with each exec.  In general the Photon ones I think keep the CPU ones and add their own but we need to verify that, especially if we use those metrics for other heuristics.